bin/grep.py

```diff
@@ -3,22 +3,78 @@
 
 from __future__ import print_function
 
-import argparse
-import collections
-import fileinput
 import os
+import io
 import re
 import sys
+import argparse
+import fileinput
+import fnmatch
+from six import PY3
+
+if PY3:
+    from pathlib import Path
+else:
+    from pathlib2 import Path
```

Comment:
IIRC, `pathlib2` is not included with pythonista (nor python 2) by default, so we should avoid using it in StaSh if possible.
The more common and simple commands (like `grep` in this case) should not have to install additional packages. As StaSh is internally recycling code often having commands call each other, requiring dependencies for such commands may result in `pip` possibly using such a command and thus requiring `pathlib2` to be installed in order to install `pathlib2`.

If any package does need external dependencies (e.g. `git`), you should:

1. Try to import the package
2. If the import fails, tell the user that the package is missing and ask if it should be installed
3. If the user confirms, call `_stash("pip install ...")`
4. Otherwise, exit with an error code.

Reply:
Very good point! I'll make sure I get the requirement of pathlib2 removed.